### PR TITLE
Made feature read and write csv files

### PR DESCRIPTION
**Pull request by Ovikt0r**

### CODE REVIEW

File: src/main/java/core/basesyntax/dao/StoreCsvDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+    private void writeLinesToFile(String lines, String pathToFile) {
+        Optional.ofNullable(lines)
+                .orElseThrow(() -> new IllegalArgumentException(STRING_ARGUMENT_ERROR));
+        try (OutputStream outputStream = new FileOutputStream(pathToFile);
+                OutputStreamWriter outputStreamWriter = new OutputStreamWriter(outputStream);
+                BufferedWriter writer = new BufferedWriter(outputStreamWriter)) {
+            writer.write(lines);
+            writer.flush();
+            System.out.println("Report data was written successfully!");
```
Comment (okuzan) [inline src/main/java/core/basesyntax/dao/StoreCsvDaoImpl.java:144]: don't forget to remove such `print`s before committing your HW

File: src/main/java/core/basesyntax/dao/StoreCsvDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        fruitTransaction.setQuantity(Integer.parseInt(row[2]));
+        return fruitTransaction;
+    }
+
+    private void writeLinesToFile(String lines, String pathToFile) {
+        Optional.ofNullable(lines)
+                .orElseThrow(() -> new IllegalArgumentException(STRING_ARGUMENT_ERROR));
+        try (OutputStream outputStream = new FileOutputStream(pathToFile);
+                OutputStreamWriter outputStreamWriter = new OutputStreamWriter(outputStream);
+                BufferedWriter writer = new BufferedWriter(outputStreamWriter)) {
```
Comment (okuzan) [inline src/main/java/core/basesyntax/dao/StoreCsvDaoImpl.java:141]:
```suggestion
        try (BufferedWriter writer = new BufferedWriter(new FileReader(pathToFile))) {
```

File: src/main/java/core/basesyntax/dao/StoreCsvDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+    private FruitTransaction getTransactionFromCsv(String line) {
+        Optional.ofNullable(line)
+                .orElseThrow(() -> new IllegalArgumentException(STRING_ARGUMENT_ERROR));
+        String[] row = line.split(",");
+        Operation operation = Operation.chooseOperation(row[0]);
+        FruitTransaction fruitTransaction = new FruitTransaction();
+        fruitTransaction.setOperation(operation);
+        fruitTransaction.setFruit(row[1]);
+        fruitTransaction.setQuantity(Integer.parseInt(row[2]));
```
Comment (okuzan) [inline src/main/java/core/basesyntax/dao/StoreCsvDaoImpl.java:132]: extract to OPERATION_INDEX etc

File: src/main/java/core/basesyntax/dao/StoreCsvDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+    private boolean isEmptyFile() {
+        boolean isEmptyFile;
+        try (InputStream inputStream = new FileInputStream(PATH_TO_DAILY_ACTIVITY_FILE);
+                InputStreamReader inputStreamReader = new InputStreamReader(inputStream,
+                        StandardCharsets.UTF_8);
+                BufferedReader reader = new BufferedReader(inputStreamReader)) {
+            isEmptyFile = reader.readLine() == null;
+        } catch (IOException e) {
+            throw new RuntimeException("Cannot read file" + PATH_TO_DAILY_ACTIVITY_FILE, e);
```
Comment (okuzan) [inline src/main/java/core/basesyntax/dao/StoreCsvDaoImpl.java:119]:
```suggestion
            throw new RuntimeException("Cannot read file " + PATH_TO_DAILY_ACTIVITY_FILE, e);
```

File: src/main/java/core/basesyntax/dao/StoreCsvDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class StoreCsvDaoImpl implements StoreCsvDao {
```
Comment (okuzan) [inline src/main/java/core/basesyntax/dao/StoreCsvDaoImpl.java:22]: it's a weird class with a weird name TBH :)) What is CSV Dao? Deal with file reading, data processing, report generation, writing to a file separately

File: src/main/java/core/basesyntax/service/impl/ReportServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        storeCsvDao.saveReportToFile(report);
+    }
+
+    private Map<String, Integer> prepareDataForReport() {
+        Map<String, Integer> fruitMap = new HashMap<>();
+        List<FruitTransaction> dailyActivities = storeCsvDao.getAll();
+        for (FruitTransaction fruitTransaction : dailyActivities) {
+            Optional.ofNullable(fruitTransaction)
+                    .orElseThrow(()
+                            -> new RuntimeException(ARGUMENT_IS_NULL_ERROR_MESSAGE
```
Comment (okuzan) [inline src/main/java/core/basesyntax/service/impl/ReportServiceImpl.java:46]: better throw more specific NoSuchElementException

**Review by okuzan — CHANGES_REQUESTED** [review]
Good job overall, nice usage of streams 👍 I believe the separation of concerns isn't clear enough here, some services are too big. Please format all of your files too (+ remove redundant empty lines)

File: src/main/java/core/basesyntax/dao/StoreCsvDaoImpl.java (hunk 2 above)
Comment (Ovikt0r) [reply src/main/java/core/basesyntax/dao/StoreCsvDaoImpl.java:141]: BufferedWriter must receive an object (heirs) of the Writer class in the constructor argument, but in this case the argument is an instance of the FileReader class. so it doesn't work. Or I don't know how to refactor it.

**Review by Vasul-Pankiv — APPROVED** [review]

**Pull request closed without merging**